config.js

```diff
@@ -542,55 +419,127 @@ exports.updateFromArgv = function() {
+        ["banunranked", "bansunranked"],
+        ["boardsize", "boardsizes"],
+        ["boardsizeranked", "boardsizesranked"],
+        ["boardsizeunranked", "boardsizesunranked"],
+        ["komi", "komis"],
+        ["komiranked", "komisranked"],
+        ["komiunranked", "komisunranked"],
+        ["speed", "speeds"],
+        ["speedranked", "speedsranked"],
+        ["speedunranked", "speedsunranked"],
+        ["timecontrol", "timecontrols"],
+        ["timecontrolranked", "timecontrolsranked"],
+        ["timecontrolunranked", "timecontrolsunranked"]
+        ];
+    for (let [oldName, newName] of deprecatedArgv) {
+        if (optimistArgv[oldName]) {
+            console.log(`Deprecated: --${oldName} is no longer supported, use --${newName} instead.`);
+        }
     }
-    if (argv.rejectnewmsg) {
-        exports.REJECTNEWMSG = argv.rejectnewmsg;
+    for (let komisGRUArg of familyArrayNamesGRU(komisFamilyNameString)) {
+        if (optimistArgv[komisGRUArg]) {
+            for (let komi of ["auto","null"]) {
+                if (optimistArgv[komisGRUArg].split(",").includes(komi)) {
+                    console.log(`Deprecated: --${komisGRUArg} /${komi}/ is no longer supported, use --${komisGRUArg} /automatic/ instead`);
+                }
+            }
+        }
     }
+    console.log("\n");
+}
 
-    exports.bot_command = argv._;
 
-    function familyArrayFromGeneralArg(generalArg) {
-        return ["", "unranked", "ranked" ].map(e => generalArg + e);
-    }
+// optimist.argv.arg(general/ranked/unranked) to exports.(r_u).arg:
+function familyArrayNamesGRU(familyNameString) {
+    return ["", "ranked", "unranked"].map(e => `${familyNameString}${e}`);
+}
 
-    function parseMinmaxRankFromNameString(rankArgNameString) {
-        let re = /(\d+)([kdp])/;
-        let results = argv[rankArgNameString].toLowerCase().match(re);
+function parseRank(arg) {
+    if (arg) {
+        const re = /(\d+)([kdp])/;
+        const results = arg.toLowerCase().match(re);
 
         if (results) {
             if (results[2] === "k") {
-                exports[rankArgNameString] = 30 - parseInt(results[1]);
+                return (30 - parseInt(results[1]));
             } else if (results[2] === "d") {
-                exports[rankArgNameString] = 30 - 1 + parseInt(results[1]);
+                return (30 - 1 + parseInt(results[1]));
             } else if (results[2] === "p") {
-                exports[rankArgNameString] = 36 + parseInt(results[1]);
-                exports.proonly = true;
-            } else {
-                console.error(`Invalid ${rankArgNameString} ${argv[rankArgNameString]}`);
-                process.exit();
+                return (36 + parseInt(results[1]));
             }
         } else {
-            console.error(`Could not parse ${rankArgNameString} ${argv[rankArgNameString]}`);
-            process.exit();
+                console.error(`error : could not parse rank -${arg}-`);
```

Comment:
Better formatting to use ```console.error(`error: could not parse rank -${arg}-`)``` with no space before the `:`.

Reply:
yes indeed :)
will fix :)